[ADD] events: attendees reports doc (17) (#9439)
odoo/documentation: content/applications/marketing/events/attendees_reports.rst

=================
Attendees reports
=================

The Odoo *Events* application allows users to create custom reports, based on the analytics and data
collected related to events. These reports can either be focused on the *Attendees* or the

> *Revenues*. Both reporting pages offer their own unique array of metrics and options.

Author: Great minds think alike! I already planned to do that, but, like you've already mentioned, I am going to wait until that is published...

:icon:`oi-view-pivot` :ref:`pivot <attendees_reports/pivot>` table, :icon:`oi-view-kanban`
:ref:`Kanban <attendees_reports/kanban>`, :icon:`oi-view-list` :ref:`list <attendees_reports/list>`,
or :icon:`oi-view-cohort` :ref:`cohort <attendees_reports/cohort>`.

These views are accessible, via their respective icons, located in the upper-right corner of the
:guilabel:`Attendees` reporting page, and have the same options as other
:doc:`reporting <../../essentials/reporting>` pages, with the same :doc:`search, filter, and group
by <../../essentials/search>` functionality.

.. note::
   The ability to :ref:`export data <export-data>` is **only** available in the :ref:`pivot
   <attendees_reports/pivot>` and :ref:`cohort <attendees_reports/cohort>` views.

.. _attendees_reports/graph:

Graph view
----------

The :guilabel:`Attendees` reporting page appears in the graph view, with the bar chart view option,
by default.

.. image:: attendees_reports/default-attendees-reporting-page.png
   :align: center
   :alt: The default bar chart in graph view of the Attendees reporting page in the Odoo Events app.

> In the :guilabel:`Search...` bar, there is also some default filters and grouping selections. The

Reviewer: ```suggestion
In the :guilabel:`Search...` bar, there are also some default filters and grouping selections. The
```

filter options specifically focus on attendee data :guilabel:`Taken` in the :guilabel:`Last 30
days`, and groups them by :guilabel:`Event > Status > Registration Date: Day`.

.. note::
   Unless manually removed from the :guilabel:`Search...` bar, this default filter is applied to

>    three other view option available on the :guilabel:`Attendees` page: pivot, Kanban, and list.

Reviewer: ```suggestion
   three other view options available on the :guilabel:`Attendees` page: pivot, Kanban, and list.
```

Kanban stage, with each attendee represented by their own Kanban card beneath their respective
event.

.. image:: attendees_reports/default-kanban-view.png
   :align: center
   :alt: The default Kanban view of the Attendees reporting page in the Odoo Events app.

Each attendee Kanban card shows their name, who they were booked by, the type of ticket they
purchased (if applicable), an indicator if the ticket was :guilabel:`Sold` or :guilabel:`Free`,
along with two clickable icon buttons: a :icon:`fa-check` :guilabel:`(checkmark)` icon and a
:icon:`fa-undo` :guilabel:`(undo)` icon.

When the :icon:`fa-check` :guilabel:`(checkmark)` icon is clicked, the attendee is manually marked
as :guilabel:`Attended`.

If the attendee was accidentally marked as :guilabel:`Attended`, click the :icon:`fa-undo`
:guilabel:`(undo)` icon to revert that action, which makes the :icon:`fa-check`
:guilabel:`(checkmark)` icon reappear on the attendee Kanban card.

Like with other reporting pages, the usual :doc:`search, filter, and group by
<../../essentials/search>` functionality can be utilized for more customized organization, if
needed.

.. example::

>    In Kanban view on the :guilabel:`Attendees` reporting page, a user could opt to remove all the
>    default filters and groupings from the :guilabel:`Search...` bar, and click the
>    :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to reveal the mega menu of
>    :guilabel:`Filters` and :guilabel:`Group By` options.
> 
>    From that mega menu, the user could opt to select :guilabel:`Ongoing Events` from the
>    :guilabel:`Filters` column. Then, they could select :guilabel:`Status` from the :guilabel:`Group
>    By` column.
> 
>    With those configurations in place, the :guilabel:`Attendees` reporting page would organize to
>    show two columns: :guilabel:`Attended` and :guilabel:`Registered`. This type of configuration
>    would be useful to see how many people are showing up (and expected to show up) for an event that
>    is currently taking place.

Reviewer 2: ```suggestion
   The report can be modified to show a company how many people are showing up (and expected to show up) for an event that
   is currently taking place. To display this information, make the following adjustments to the report. 
   
   In Kanban view on the :guilabel:`Attendees` reporting page, remove all the
   default filters and groupings from the :guilabel:`Search...` bar, and click the
   :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to reveal the mega menu of
   :guilabel:`Filters` and :guilabel:`Group By` options.

   From that mega menu, select :guilabel:`Ongoing Events` from the
   :guilabel:`Filters` column. Then, select :guilabel:`Status` from the :guilabel:`Group
   By` column.

   With those configurations in place, the :guilabel:`Attendees` reporting page displays two columns: :guilabel:`Attended` and :guilabel:`Registered`. 
```

Same suggestion as above- to introduce the idea/what the report is showing, then a more active language to explain ow to achieve that report

attendee data :guilabel:`Taken` in the :guilabel:`Last 30 days`, and groups them by :guilabel:`Event
> Status > Registration Date: Day`.

With those default filters and grouping in place, the data is presented in a nested list format,
with each event in the database revealing a nested sub-list that either contains the data for people
who have :guilabel:`Attended` or :guilabel:`Registered` for the event.

.. image:: attendees_reports/default-list-view.png
   :align: center
   :alt: The default list view of the Attendees reporting page in the Odoo Events app.

By default, the list view of the :guilabel:`Attendees` page showcases the following information in columns:
:guilabel:`Registration Date`, :guilabel:`Attendee Name`, :guilabel:`Email`, :guilabel:`Phone`,
:guilabel:`Event`, :guilabel:`Event Ticket`, :guilabel:`Activities` (if applicable),
:guilabel:`Status`, and :guilabel:`Sale Status`.

To add (or remove) columns from the list view of the :guilabel:`Attendees` reporting page, click the
:icon:`oi-settings-adjust` :guilabel:`(additional options)` icon, located to the far-right of the

> column names to reveal a drop-down menu of additional column options to choose from.

Reviewer: ```suggestion
column names, to reveal a drop-down menu of additional column options to choose from.
```

When the :icon:`fa-caret-right` :guilabel:`(right arrow)` beside a row title is clicked, the
sub-list for that row is revealed below. To re-tuck the list, click the subsequent
:icon:`fa-caret-down` :guilabel:`(down arrow)` beside the desired row title.

Like with other reporting pages, the usual :doc:`search, filter, and group by
<../../essentials/search>` functionality can be utilized for more customized organization, if
needed.

.. example::

>    After removing all the default filters and groupings from the :guilabel:`Attendees` reporting
>    page, while in list view, the user could click the :icon:`fa-caret-down` :guilabel:`(down arrow)`
>    to the right of the :guilabel:`Search...` bar to reveal the mega menu of :guilabel:`Filters` and
>    :guilabel:`Group By` options.
> 
>    From this mega menu, the user could choose to select the :guilabel:`Registration Date`
>    :icon:`fa-caret-down` drop-down menu, and select a desired period of time to analyze: a month, a
>    quarter, or a year. In this case, they choose the current month, :guilabel:`May`. Doing so also
>    selects the current year, as well.
> 
>    Then, the user could opt to choose the :guilabel:`Partner`, :guilabel:`Event`, :guilabel:`Ticket
>    Type`, and :guilabel:`Status` from the :guilabel:`Group By` column.
> 
>    With that filter and groupings in place, Odoo presents the nested list of attendee data in such a
>    way that organizes the analytics from the current month, first by :guilabel:`Partner`, then by
>    :guilabel:`Event`, followed by :guilabel:`Ticket Type`, and lastly, by :guilabel:`Status`.

Reviewer 2: ```suggestion
A company can modify this list to analyze which partners are registering the most attendees for events over a specific period of time (a month, quarter, or year). In this example, the data is analyzed for the current month (May), and is further organized by event and ticket type. To view a report with these parameters, the following configurations must be made. 
   
   While in list view, remove all the default filters and groupings from the :guilabel:`Attendees` reporting
   page. Then, click the :icon:`fa-caret-down` :guilabel:`(down arrow)`
   to the right of the :guilabel:`Search...` bar to reveal the mega menu of :guilabel:`Filters` and
   :guilabel:`Group By` options.

   From this mega menu, select the :guilabel:`Registration Date`
   :icon:`fa-caret-down` drop-down menu, and select the current month, :guilabel:`May`. Doing so also selects the current year, as well.

   Then, choose the :guilabel:`Partner`, :guilabel:`Event`, :guilabel:`Ticket
   Type`, and :guilabel:`Status` from the :guilabel:`Group By` column.
```

Same suggestions as before- also optional.

.. _attendees_reports/cohort:

Cohort view
-----------

Click the :icon:`oi-view-cohort` :guilabel:`cohort` icon in the upper-right corner to view the
:guilabel:`Attendees` reporting metrics in a cohort table.

By default, the cohort view **only** has the :guilabel:`Taken` and :guilabel:`Last 30 days` filter
options present in the :guilabel:`Search...` bar, by default.

The data displayed in the cohort table is organized by :guilabel:`Day`, by default.

It should also be noted that the only option available in the :guilabel:`Measures` drop-down menu is
:guilabel:`Count`, which is selected, by default.

.. image:: attendees_reports/default-cohort-view.png
   :align: center
   :alt: The default cohort view of the Attendees reporting page in the Odoo Events app.

On the far-left, the column shows the :guilabel:`Event Start Date`, and next to the that is the
default :guilabel:`Count` measure option.

> To the right of those columns is the body of the cohort table, showcasing the :guilabel:`Create on -

Reviewer 2: ```suggestion
To the right of those columns is the body of the cohort table, showcasing the :guilabel:`Created on -
```

(selected time period)` data, which is :guilabel:`Day`, by default. The numbered columns below that
show the days (via negative integers) leading up to the start of the event (represented by the
:guilabel:`0` to the far-right).

Beneath each of those days leading up to the start of the event, there is a percentage indicating

> how many tickets were :guilabel:`Taken` on that particular day (per the default settings present in

Reviewer 2: ```suggestion
how many tickets were :guilabel:`Taken` (or registered) on that particular day (per the default settings present in
```
I don't know if that is what the metric means- if so, it might be good to mention to the user what 'taken' is implying. I keep picturing a theater usher 'taking' my ticket and ripping it as I walk in- but these are registrants before the "0" day, so I assume it's registered. Just an idea!

the :guilabel:`Search...` bar).

> .. example::

Reviewer 2: AWESOME example!

   If an :guilabel:`Event Start Date` is May 17th, and has a :guilabel:`Count` of :guilabel:`3`,
   that means three total tickets were sold for that event, prior to it starting.

   If one of those three tickets was sold on May 13th (four days prior to the :guilabel:`Event Start

>    Date`), a percentage of `33%` appears in the `-4` column for that event -- indicating that one of

Reviewer: ```suggestion
   Date`), a percentage of `33%` appears in the `-4` column for that event, indicating that one of
```
